app/controllers/ops_controller/settings/schedules.rb

```diff
@@ -545,74 +546,117 @@ def build_attrs_from_params(params)
     { MiqAeEngine.create_automation_attribute_key(object).to_s => MiqAeEngine.create_automation_attribute_value(object) }
   end
 
-  def build_search_filter_from_params
-    case params[:action_typ]
+  def parse_filter_value(filter_value)
+    values      = filter_value&.split("__")
+    value       = values&.first
+    other_value = values&.size == 1 ? "" : values&.last
+
+    [value, other_value]
+  end
+
+  def convert_filter_type_to_klass(filter_type)
+    case filter_type
+    when "ems"
+      "ExtManagementSystem"
+    when "cluster"
+      "EmsCluster"
+    when "host"
+      "Host"
     when "storage"
-      case params[:filter_typ]
-      when "ems"     then {"CONTAINS" => {"field" => "Storage.ext_management_systems-name", "value" => params[:filter_value]}}
-      when "host"    then {"CONTAINS" => {"field" => "Storage.hosts-name", "value" => params[:filter_value]}}
-      when "storage" then {"=" => {"field" => "Storage-name", "value" => params[:filter_value]}}
-      else                {"IS NOT NULL" => {"field" => "Storage-name"}}
+      "Storage"
+    when "vm"
+      "Vm"
+    when "miq_template"
+      "MiqTemplate"
+    else
+      "Base"
+    end
+  end
+
+  def build_filter_expression_from(schedule, filter_type, value, other_value)
```

Comment:
I'd vote for renaming this routine name to `build_filter_expression`. Since we're returning a new `MiqExpression` here, we're building an actual expression here and the `_from` at the end doesn't quite make sense.

Reply:
Fixed, thanks 👍